src/components/export-gist.jsx

```diff
@@ -0,0 +1,41 @@
+/* globals XMLHttpRequest:false window:false */
+import React from "react";
+
+export default class ExportGist extends React.Component {
+  postGist() {
+    const {containerElement} = this.props;
+
+    const request = new XMLHttpRequest();
+    request.onreadystatechange = () => {
+      if (request.readyState === 4 && request.status === 201) {
+        const response = JSON.parse(request.responseText);
```

Comment:
Add error handling:
- Wrap `JSON.parse` in a `try-catch` statement.
- Also add a case for `request.status >== 400`.
- Suggestion: display an error message in the rendered component on error.